contracts/crosschain/axelar/AxelarGatewayAdapter.sol

```diff
@@ -0,0 +1,179 @@
+        if (addr.length == 0)
+            revert UnsupportedERC7930Chain(InteroperableAddress.formatV1(chainType, chainReference, ""));
+        return addr;
+    }
+
+    /// @dev Registers a chain equivalence between a binary interoperable chain id and an Axelar chain identifier.
+    function registerChainEquivalence(bytes calldata chain, string calldata axelar) public virtual onlyOwner {
+        (, , bytes calldata addr) = chain.parseV1Calldata();
+        require(addr.length == 0, InvalidChainIdentifier(chain));
+        require(
+            bytes(_erc7930ToAxelar[chain]).length == 0 && _axelarToErc7930[axelar].length == 0,
+            ChainEquivalenceAlreadyRegistered(chain, axelar)
+        );
+
+        _erc7930ToAxelar[chain] = axelar;
+        _axelarToErc7930[axelar] = chain;
+        emit RegisteredChainEquivalence(chain, axelar);
+    }
+
+    /// @dev Registers the address of a remote gateway.
+    function registerRemoteGateway(bytes calldata remote) public virtual onlyOwner {
+        (bytes2 chainType, bytes calldata chainReference, bytes calldata addr) = remote.parseV1Calldata();
+        require(
+            _remoteGateways[chainType][chainReference].length == 0,
+            RemoteGatewayAlreadyRegistered(chainType, chainReference)
+        );
+        _remoteGateways[chainType][chainReference] = addr;
+        emit RegisteredRemoteGateway(remote);
+    }
+
+    /// @inheritdoc IERC7786GatewaySource
+    function supportsAttribute(bytes4 /*selector*/) public pure returns (bool) {
+        return false;
+    }
+
+    /// @inheritdoc IERC7786GatewaySource
+    function sendMessage(
+        bytes calldata recipient, // Binary Interoperable Address
+        bytes calldata payload,
+        bytes[] calldata attributes
+    ) external payable returns (bytes32) {
+        require(msg.value == 0, UnsupportedNativeTransfer());
+        // Use of `if () revert` syntax to avoid accessing attributes[0] if it's empty
+        if (attributes.length > 0)
+            revert UnsupportedAttribute(attributes[0].length < 0x04 ? bytes4(0) : bytes4(attributes[0][0:4]));
+
+        // Create the package
+        bytes memory sender = InteroperableAddress.formatEvmV1(block.chainid, msg.sender);
+        bytes memory adapterPayload = abi.encode(sender, recipient, payload);
+
+        // Emit event early (stack too deep)
+        bytes32 sendId = bytes32(0); // Explicitly set to 0
+        emit MessageSent(sendId, sender, recipient, payload, 0, attributes);
+
+        // Send the message
+        (bytes2 chainType, bytes calldata chainReference, ) = recipient.parseV1Calldata();
+        bytes memory remoteGateway = getRemoteGateway(chainType, chainReference);
+        string memory axelarDestination = getAxelarChain(InteroperableAddress.formatV1(chainType, chainReference, ""));
+        string memory axelarTarget = address(bytes20(remoteGateway)).toChecksumHexString(); // TODO non-evm chains?
```

Comment:
Let's remove all "TODO" comments. If the contract doesn't support non-EVM chains, let's document that in the contract-level documentation.